tools/mtouch/Application.cs

```diff
@@ -1416,14 +1416,33 @@ void GenerateMSymManifest (Target target, string target_directory)
 			Driver.WriteIfDifferent (manifestPath, root.ToString ());
 		}
 
-		void CopyAssemblyData (Assembly asm, string targetDir)
+		void CopyAotData (string src, string dest)
 		{
-			var mvid = asm.AssemblyDefinition.MainModule.Mvid.ToString ().ToUpperInvariant ();
-			var assemblyDir = Path.Combine (targetDir, mvid);
-			if (!Directory.Exists (assemblyDir))
-				Directory.CreateDirectory (assemblyDir);
-			asm.CopyToDirectory (assemblyDir, reload: false, only_copy: true);
-			asm.CopyMSymToDirectory (targetDir);
+			if (string.IsNullOrEmpty (src) || string.IsNullOrEmpty (dest)) {
+				ErrorHelper.Warning (95, $"Aot files could not be copied to the destination directory {dest}"); 
+				return;
+			}
+				
+			var dir = new DirectoryInfo (src);
+			if (!dir.Exists) {
+				ErrorHelper.Warning (95, $"Aot files could not be copied to the destination directory {dest}"); 
+				return;
+			}
+
+			var dirs = dir.GetDirectories ();
```

Comment:
That can be fixed later :-)
Let's try to make it work first (then backport to `cycle8`).